crates/hir-def/src/path/lower.rs

```diff
@@ -181,7 +184,10 @@ pub(super) fn lower_generic_args(
                 }
             }
             ast::GenericArg::ConstArg(arg) => {
-                let arg = ConstScalarOrPath::from_expr_opt(arg.expr());
+                let arg = arg.expr().map_or(
```

Comment:
is this just inlining from_expr_opt? let's keep that unless there's a good reason for this change.

Reply:
@flodiebold I've read [this](https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/style.md#function-preconditions) and decided function accepting `Option<_>` is bad

Reply:
That's not the intention of the guideline. This is not a precondition, the `from_expr_opt` function is specifically for handling missing exprs.